util/types/datum.go

```diff
@@ -1519,6 +1502,37 @@ func (d *Datum) ToBytes() ([]byte, error) {
 	}
 }
 
+// ToMysqlJSON is similar to convertToMysqlJSON, except the
+// latter parses from string, but the former uses it as primitive.
+func (d *Datum) ToMysqlJSON() (j json.JSON, err error) {
+	switch d.Kind() {
+	case KindNull:
+		j = json.CreateJSON(nil)
+	case KindMysqlJSON:
+		j = d.x.(json.JSON)
+	case KindInt64, KindUint64:
+		i64 := d.GetInt64()
+		j = json.CreateJSON(i64)
+	case KindFloat32, KindFloat64:
+		f64 := d.GetFloat64()
+		j = json.CreateJSON(f64)
+	case KindMysqlDecimal:
+		f64, _ := d.GetMysqlDecimal().ToFloat64()
+		j = json.CreateJSON(f64)
+	case KindString, KindBytes:
+		s := d.GetString()
+		j = json.CreateJSON(s)
+	default:
+		s, convErr := d.ToString()
+		if convErr != nil {
+			err = errors.Trace(convErr)
+			return
+		}
+		j = json.CreateJSON(s)
```

Comment:
Seems `json.CreateJSON` are called in all cases, could it be put before return and write only once?